zerver/views/auth.py

```diff
@@ -719,6 +750,22 @@ def login_page(
     # that disables the default behavior of redirecting logged-in users to the
     # logged-in app.
     is_preview = 'preview' in request.GET
+
+    extra_context = kwargs.pop('extra_context', {})
+    extra_context["next"] = next
+
+    if extra_context.get("action_key") is None:
+        extra_context["action_key"] = ''
+    else:
+        # We want to authenticate user while letting to be logged in.
+        is_preview = True
+
+    action_key_from_post = request.POST.get('action_key')
+    action_key_exists_in_post  = action_key_from_post and action_key_from_post is not None
+    if action_key_exists_in_post:
```

Comment:
This seems redundant, all we need is
```suggestion
    if action_key_from_post:
```
since it does exactly thhe same

Reply:
Ohh, yes, it's redundant. I was a bit confused here when puppeteer test failed and ended up doing this. changed it.